app/controllers/application_controller.rb

```diff
@@ -381,24 +381,22 @@ def get_viewers_for(data, data_type)
 
   def viewers_hover(data, link)
     result = ''
-    viewers = ''
-
-    if link
-      viewers += t('shared.viewers_hover.visible_to')
-    end
 
     if data.blank? || data.length == 0
       if link
-        viewers += t('shared.viewers_hover.only_you').downcase
+        viewers = t('shared.viewers.you_link')
       else
-        viewers += t('shared.viewers_hover.only_you')
+        viewers = t('shared.viewers.you')
+      end
+    else
+      viewer_names = data.to_a.map { |user_id| User.find(user_id).name }.to_sentence()
+      if link
+        viewers = t('shared.viewers.many', viewers: viewer_names)
+      else
+        viewers = viewer_names
```

Comment:
I started working on a refactor in another branch, if you want to check it out - [controller_cleanup](https://github.com/julianguyen/ifme/tree/controller_cleanup